src/Containers/Microsoft.NET.Build.Containers/Registry/DefaultRegistryAPI.cs

```diff
@@ -71,4 +55,40 @@ private static HttpClient CreateClient(string registryName, Uri baseUri, ILogger
 
         return client;
     }
+
+    private static HttpMessageHandler CreateHttpHandler(Uri baseUri, bool allowInsecure, ILogger logger)
+    {
+        var socketsHttpHandler = new SocketsHttpHandler()
+        {
+            UseCookies = false,
+            // the rest of the HTTP stack has an very long timeout (see below) but we should still have a reasonable timeout for the initial connection
+            ConnectTimeout = TimeSpan.FromSeconds(30)
+        };
+
+        if (!allowInsecure)
+        {
+            return socketsHttpHandler;
+        }
+
+        socketsHttpHandler.SslOptions = new System.Net.Security.SslClientAuthenticationOptions()
+        {
+            RemoteCertificateValidationCallback = (object sender, X509Certificate? certificate, X509Chain? chain, SslPolicyErrors sslPolicyErrors) =>
+            {
+                if (sslPolicyErrors == SslPolicyErrors.None)
+                {
+                    return true;
+                }
+
+                // Ignore certificate errors for the hostname.
+                if ((sender as SslStream)?.TargetHostName == baseUri.Host)
+                {
+                    return true;
+                }
+
+                return false;
+            }
+        };
```

Comment:
This is readable as-is, but would you mind extracting it out to a named function - maybe something like "IgnoreCertificateErrorsForSpecificHost" - to try and help future-us more easily remember the meaning?